### PR TITLE
 Convert to use shared AppVeyor module and Markdown/Example Tests - Fixes #68

### LINKED ISSUE #68: xComputer JoinDomain fails when server is already in domain
Resource fails with fatal error if the computer already is in the domain.
The test should pick this up, which now it doesn't and hence throws a fatal exception when attempting to join the domain again.

**Error:** 
Cannot add computer 'x' to domain 'y' because it is already in that domain.
    + CategoryInfo          : InvalidOperation: (x:) [], CimException
    + FullyQualifiedErrorId : AddComputerToSameDomain,Microsoft.PowerShell.Commands.AddComputerCommand
    + PSComputerName        : localhost

### PR DESCRIPTION
**Pull request by PlagueHO**
This PR converts the resource module to use the shared AppVeyor.psm1 module from DSCResource.Tests. It also fixes the README.MD markdown errors. Support for codecoverage reporting via CodeCov.io has also been added.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xcomputermanagement/77)

### CODE REVIEW

**Comment by PlagueHO** [comment]
Hi @RichardSiddaway - I thought I'd tag you for a review on this one because I noticed you're marked as a repo maintainer now :grin: and I've been hammering the other repo maintainers with review requests pretty hard lately.

Any chance you take a look at this one? It is primarily a clean up job to try and **begin** to bring xComputerManagement up to standards.

I'm think it is OK that we're at 68% unit test coverage, but I'll try and bring it up to the magic 70% mark over the next few months.

File: appveyor.yml [hunk 1]
```diff
@@ -23,43 +21,15 @@ build: false
 
 test_script:
     - ps: |
-        $testResultsFile = ".\TestsResults.xml"
-        $res = Invoke-Pester -OutputFormat NUnitXml -OutputFile $testResultsFile -PassThru
-        (New-Object 'System.Net.WebClient').UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path $testResultsFile))
-        if ($res.FailedCount -gt 0) {
-            throw "$($res.FailedCount) tests failed."
-        }
+          Invoke-AppveyorTestScriptTask `
```
Comment (johlju) [inline appveyor.yml]: You need `-ExcludeTag @()` otherwise they are excluded ant you can't opt-in.

**Comment by johlju** [comment]
I just saw that while browsing. Have not done a review. Hoping @RichardSiddaway have time to the full review. 😄

**Comment by PlagueHO** [comment]
@johlju - any chance you could review this one for me if you get a moment?

**Comment by johlju** [comment]
Reviewed 3 of 6 files at r1, 1 of 2 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[.markdownlint.json, line 4 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpETnZLZeWFXeC4qh6a:-KpETnZLZeWFXeC4qh6b:b-emm32j) ([raw file](https://github.com/powershell/xcomputermanagement/blob/e65f65c5970dd93d9d6cd8eedcda75654f12da76/.markdownlint.json#L4)):*
> ```JSON
>     "default": true,
>     "MD029": {
>         "style": "ordered"
> ```

This should be "one" now. Changed in PR recently merged into DscResource.Tests.

---

*[README.md, line 10 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpEUH9elO_zbhpskDHl:-KpEUH9elO_zbhpskDHm:b-sqg70z) ([raw file](https://github.com/powershell/xcomputermanagement/blob/e65f65c5970dd93d9d6cd8eedcda75654f12da76/README.md#L10)):*
> ```Markdown
> The **xComputerManagement** module contains the following resources:
> 
> * xComputer - allows you to configure a computer by changing its name and
> ```

Maybe add links to their respectively header on each of these resources?
You don't have to do this in this PR. Okay as-is.

---

*[README.md, line 18 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpEUrQSkPV6n1ewIqWw:-KpEUrQSkPV6n1ewIqWx:b-26yawn) ([raw file](https://github.com/powershell/xcomputermanagement/blob/e65f65c5970dd93d9d6cd8eedcda75654f12da76/README.md#L18)):*
> ```Markdown
>   local computer.
> * xPowerPlan - specifies a power plan to activate.
> ```

Missing resource xVirtualMemory here

---

*[README.md, line 252 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpEW3BxKh815v_qiR54:-KpEW3BxKh815v_qiR55:bq0k04b) ([raw file](https://github.com/powershell/xcomputermanagement/blob/e65f65c5970dd93d9d6cd8eedcda75654f12da76/README.md#L252)):*
> ```Markdown
> 
> ```powershell
> configuration Sample_xComputer_ChangeNameAndWorkGroup
> ```

I started to comment on the examples, but realized there was a lot style "errors" so you have not fixed. Those. Do you want to fix these in this PR or in another PR?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77)*

**Comment by PlagueHO** [comment]
@johlju - I've borrowed the branch header from xSQLServer README.MD and I'll convert all resource I maintain over to that for consistency. Hope that is OK? Should be good to review now.

---

Review status: 4 of 6 files reviewed at latest revision, 5 unresolved discussions.

---

*[.markdownlint.json, line 4 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpETnZLZeWFXeC4qh6a:-KpHFfDqkpH7tAqFmJXo:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/e65f65c5970dd93d9d6cd8eedcda75654f12da76/.markdownlint.json#L4)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This should be "one" now. Changed in PR recently merged into DscResource.Tests.
</blockquote></details>

Done.

---

*[appveyor.yml, line 24 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KoGw-dpBGPLFAdOkOx2-r1-24:-KpHFhzkhG1_iFLggin2:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/6441f3156065860a29f19069a770796b125e1957/appveyor.yml#L24)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

You need `-ExcludeTag @()` otherwise they are excluded ant you can't opt-in.
</blockquote></details>

Done.

---

*[README.md, line 10 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpEUH9elO_zbhpskDHl:-KpHFkHZJGvTXTGlB1OO:bidspu) ([raw file](https://github.com/powershell/xcomputermanagement/blob/e65f65c5970dd93d9d6cd8eedcda75654f12da76/README.md#L10)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Maybe add links to their respectively header on each of these resources?
You don't have to do this in this PR. Okay as-is.
</blockquote></details>

Normally I'd do this, but as I'm converting this to Wiki/Auto-documentation soon (just doing the clean up in stages) it won't make much sense to do so :grin:

---

*[README.md, line 18 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpEUrQSkPV6n1ewIqWw:-KpHFvQzZkUWM9Xur3_B:bm4qzni) ([raw file](https://github.com/powershell/xcomputermanagement/blob/e65f65c5970dd93d9d6cd8eedcda75654f12da76/README.md#L18)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Missing resource xVirtualMemory here
</blockquote></details>

Good catch! Done

---

*[README.md, line 252 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpEW3BxKh815v_qiR54:-KpHG4dIn5RUebbvKrai:b-i036v9) ([raw file](https://github.com/powershell/xcomputermanagement/blob/e65f65c5970dd93d9d6cd8eedcda75654f12da76/README.md#L252)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I started to comment on the examples, but realized there was a lot style "errors" so you have not fixed. Those. Do you want to fix these in this PR or in another PR?
</blockquote></details>

I've added another issue to get this fixed :grin: I'm trying to break the clean up of this resource down into stages. These examples will all go anyway once I move to Wiki/Auto-documentation (very soon).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-:-KpHGI-8m7TqdKI1LOBw:bpqyd2x)*

**Comment by johlju** [comment]
Yes, of course it's okay :smile: Glad that it could be reused. i will review those other repos as well.  :smile: 

You got tests failing
https://ci.appveyor.com/project/PowerShell/xcomputermanagement/build/1.9.244.0#L35

---

Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[appveyor.yml, line 6 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpK2mKoplK2-1_dCv4V:-KpK2mKoplK2-1_dCv4W:bwb9k0l) ([raw file](https://github.com/powershell/xcomputermanagement/blob/0f06065880c325cfa42e48c4daa8114b2fc110c4/appveyor.yml#L6)):*
> ```YAML
> version: 1.9.{build}.0
> install:
>     - git clone -b Issue-86 https://github.com/PlagueHO/DscResource.Tests
> ```

I don't think this should be here. Point to your fork. :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-:-KpK2NoTm0yid6rImU9i:b83z5y)*

**Comment by PlagueHO** [comment]
Thanks for reviewing @Johlju - the tests will continue to fail on this one until this PR (https://github.com/PowerShell/DscResource.Tests/pull/158) goes through. Once it is in I'll trigger AppVeyor again and they should then pass.

---

Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[appveyor.yml, line 6 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-KpK2mKoplK2-1_dCv4V:-KpK6OWORZfpiKoR91Sb:b-nx5zo9) ([raw file](https://github.com/powershell/xcomputermanagement/blob/0f06065880c325cfa42e48c4daa8114b2fc110c4/appveyor.yml#L6)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I don't think this should be here. Point to your fork. :)
</blockquote></details>

Fixed. I was using this PR/repo to test the Custom PSSA rules.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-:-KpK6VB46huzVTQvpnUk:bf2bsbe)*

**Comment by johlju** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/77#-:-KpK9ZhOqCHhAjDi6T6M:bnfp4nl)*

**Comment by PlagueHO** [comment]
I'll ignore the codecov not meeting target of 70% as we're only just introducing this. This has been raised in issue #81.

**Pull request merged**